app/services/generate_vendor_providers.rb

```diff
@@ -13,10 +13,28 @@ def self.call
       { name: 'Gordon Associates Provider', code: 'GORD' },
       { name: 'University of Newcastle Upon Tyne', code: 'N21' },
       { name: 'University College Birmingham', code: 'B35' },
+      { name: 'Test Provider', code: 'TEST' },
     ]
 
     providers.each do |provider|
       Provider.find_or_create_by(provider)
     end
+
+    providers.reject { |p| p[:code] == 'TEST' }.each do |provider|
+      generate_courses_as_an_accredited_body(provider[:code])
```

Comment:
It seems like you can just pass `provider` here to `generate_courses_as_an_accredited_body` and then you won't have to `Provider.find_by(code: accredited_body_code)` in this method. (See review comment above)